test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/ProjectRestoreInfoBuilder.cs

```diff
@@ -90,12 +106,22 @@ public ProjectRestoreInfoBuilder WithTool(string name, string version)
         public ProjectRestoreInfoBuilder WithTargetFrameworkInfo(
             IVsTargetFrameworkInfo tfi)
         {
-            (_pri.TargetFrameworks as VsTargetFrameworks).Add(tfi);
+            if(_projectRestoreInfo.TargetFrameworks is VsTargetFrameworks && tfi is VsTargetFrameworkInfo)
+            {
+                (_projectRestoreInfo.TargetFrameworks as VsTargetFrameworks).Add(tfi);
+            }
+
+            if (_projectRestoreInfo2.TargetFrameworks is VsTargetFrameworks2 && tfi is VsTargetFrameworkInfo2)
+            {
+                (_projectRestoreInfo2.TargetFrameworks as VsTargetFrameworks2).Add((IVsTargetFrameworkInfo2) tfi);
+            }
 
             return this;
         }
 
-        public VsProjectRestoreInfo Build() => _pri;
+        public VsProjectRestoreInfo Build() => _projectRestoreInfo;
+
+        public VsProjectRestoreInfo2 Build2() => _projectRestoreInfo2;
```

Comment:
`System.UriBuilder` just has a property (`Uri`) which returns the constructed result.

`Build()` and `Build2()` don't actually do anything.  How about converting these to read-only properties `ProjectRestoreInfo` and `ProjectRestoreInfo2`?